crates/astria-proto/src/native/composer/block_relay/v1alpha1/bid.rs

```diff
@@ -0,0 +1,159 @@
+use ed25519_consensus::{
+    Signature,
+    SigningKey,
+    VerificationKey,
+};
+use prost::Message;
+
+use crate::{
+    generated::block_relay::v1alpha1 as raw,
+    native::sequencer::v1alpha1::{
+        TransferAction,
+        UnsignedTransaction,
+    },
+};
+
+#[derive(Debug, thiserror::Error)]
+pub enum SignedBundleError {
+    #[error("converting from proto failed")]
+    ProtoDecodeFailed,
+}
+
+pub struct SignedBundle {
+    pub signature: Signature,
+    pub verification_key: VerificationKey,
+    pub bundle: Bundle,
+}
+
+impl SignedBundle {
+    // into_raw
+    // to_raw
+    // try_from_raw
+
+    pub fn try_from_raw(raw: raw::SignedBundle) -> Result<Self, SignedBundleError> {
+        let signature = Signature::try_from(&*raw.signature)
+            .map_err(|_| SignedBundleError::ProtoDecodeFailed)?;
+        let verification_key = VerificationKey::try_from(&*raw.public_key)
+            .map_err(|_| SignedBundleError::ProtoDecodeFailed)?;
+        let bundle = Bundle::try_from_raw(raw.bundle.ok_or(SignedBundleError::ProtoDecodeFailed)?)?;
+        Ok(Self {
+            signature,
+            verification_key,
+            bundle,
+        })
+    }
+    // into_parts
+}
+
+pub struct Bundle {
+    pub block_height: u64,
+    pub bid: TransferAction,
+    pub payload: UnsignedTransaction,
+}
+
+impl Bundle {
+    // into_signed
+
+    // Convert the bundle into an `OpaqueBid` that can be sent to the Proposer by hashing the
+    // payload with SHA256.
+    pub fn to_opaque_bid(&self) -> OpaqueBid {
+        use sha2::Digest as _;
+
+        let bytes = self.payload.to_raw().encode_to_vec();
+        let mut hasher = sha2::Sha256::new();
+        hasher.update(bytes);
+        let payload_hash: [u8; 64] = hasher.finalize().into();
+
+        OpaqueBid {
+            block_height: self.block_height,
+            bid: self.bid.clone(),
+            payload_hash,
+        }
+    }
+
+    // into_raw
+    // to_raw
+    // try_from_raw
+    pub fn try_from_raw(raw: raw::Bundle) -> Result<Self, SignedBundleError> {
+        let bid =
+            TransferAction::try_from_raw(raw.bid.ok_or(SignedBundleError::ProtoDecodeFailed)?)
+                .map_err(|_| SignedBundleError::ProtoDecodeFailed)?;
+        let payload = UnsignedTransaction::try_from_raw(
+            raw.bundle.ok_or(SignedBundleError::ProtoDecodeFailed)?,
+        )
+        .map_err(|_| SignedBundleError::ProtoDecodeFailed)?;
+        Ok(Self {
+            block_height: raw.block_height,
+            bid,
+            payload,
+        })
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum OpaqueBidError {
```

Comment:
these should probably have something similar to what's done for `SignedTransactionError` and the like in [`sequencer::v1alpha1`](https://github.com/astriaorg/astria/blob/80cbef399b2d6e688671cbda887a1294a55d2ddd/crates/astria-proto/src/native/sequencer/v1alpha1.rs#L23) instead of these silly error enums